src/gui/editors/PatternEditor.cpp

```diff
@@ -46,9 +48,24 @@ namespace lmms::gui
 
 PatternEditor::PatternEditor(PatternStore* ps) :
 	TrackContainerView(ps),
-	m_ps(ps)
+	m_ps(ps),
+	m_trackHeadWidth(ConfigManager::inst()->value("ui", "compacttrackbuttons").toInt() == 1
+		? DEFAULT_SETTINGS_WIDGET_WIDTH_COMPACT + TRACK_OP_WIDTH_COMPACT
+		: DEFAULT_SETTINGS_WIDGET_WIDTH + TRACK_OP_WIDTH),
+	m_maxSteps(TimePos::stepsPerBar())
 {
 	setModel(ps);
+
+	m_timeLine = new TimeLineWidget(m_trackHeadWidth, 32, pixelsPerBar(),
+		Engine::getSong()->getPlayPos(Song::PlayMode::Pattern),
+		Engine::getSong()->getTimeline(Song::PlayMode::Pattern),
+		m_currentPosition, Song::PlayMode::Pattern, this
+	);
+	connect(m_timeLine, &TimeLineWidget::positionChanged, this, &PatternEditor::updatePosition);
+	static_cast<QVBoxLayout*>( layout() )->insertWidget( 0, m_timeLine );
```

Comment:
I'm not sure why we need the `static_cast` here?
```suggestion
	layout()->insertWidget(0, m_timeLine);
```

Reply:
```
error: ‘class QLayout’ has no member named ‘insertWidget’; did you mean ‘parentWidget’?
   65 |         layout()->insertWidget(0, m_timeLine);
      |                   ^~~~~~~~~~~~
      |                   parentWidget
```
I think it has to be cast to a vbox in order for it to work.